lucene/core/src/java/org/apache/lucene/util/automaton/SortedIntSet.java

```diff
@@ -149,128 +156,32 @@ public void computeHash() {
         hashCode = 683*hashCode + values[i];
       }
     }
+    stale = false;
   }
 
+  /**
+   * Create a FrozenIntSet from the current values in this IntSet.
+   *
+   * @return a FrozenIntSet that has the same values and hashCode as this set
+   */
+  @SuppressForbidden(reason = "Arrays.copyOf is a candidate for JVM intrinsic and we know our bounds are good")
```

Comment:
Hmm why did we mark this API forbidden again?

Reply:
Because copyOf was considered trappy -- the new length can be larger than original size of the input array (and is then null or zero padded)... I'm not sure it's worth having System.arraycopy instead... Maybe we could have a utility method that assert-verifies newLength <= length and uses Arrays.copyOf. Not sure.